### DISCUSSION TITLE
Guide/example for a simple C module

### DISCUSSION THREAD

**Discussion by yukijoou** [fcitx/fcitx5#825] [General]
Hello,

I would like to write a module for fcitx in C that detects when the user focuses an input field. I tried looking around, but couldn't find easy to follow documentation or simple code examples. Is there such a thing as a "bare minimum module" that I could look at to base my work on?

I did find [this page](https://fcitx.github.io/developer-handbook/fcitx-dev.html) of the developer handbook, but it only has sparse bits of code, so it's hard to get an idea of what a full module looks like.

**Comment by wengxt** [comment]
That link is for fcitx 4, not very helpful as for fcitx5.

You may want to check https://fcitx-im.org/wiki/Develop_an_simple_input_method

Input method itself is just a special type of addon, most information may still apply if you just want a module.

**Reply by wengxt** [reply]
As for monitor focus change, you'll need this function:
https://codedocs.xyz/fcitx/fcitx5/classfcitx_1_1Instance.html#a19894a928b3b511422682609dcfbe4b7

Type would be InputContextFocusIn  https://codedocs.xyz/fcitx/fcitx5/event_8h_source.html

**Reply by yukijoou** [reply]
Thank you very much for your help ^_^

I think it would be a good idea to add a deprecation warning to the Fcitx4 documentation, since it seems to come up much higher on search engines than the fcitx5 documentation…

**Reply by wengxt** [reply]
Thanks, I add the redirection to wiki instead.
